pkg/nemesis/scaling.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nemesis
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+
+	clusterTypes "github.com/pingcap/tipocket/pkg/cluster/types"
+	"github.com/pingcap/tipocket/pkg/core"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type scalingGenerator struct {
+	name string
+}
+
+func NewScalingGenerator(name string) core.NemesisGenerator {
+	return scalingGenerator{name: name}
+}
+
+func (s scalingGenerator) Generate(nodes []clusterTypes.Node) []*core.NemesisOperation {
+	ops := make([]*core.NemesisOperation, 1)
+	ops[0] = &core.NemesisOperation{
+		Type:        core.Scaling,
+		Node:        &nodes[0],
+		InvokeArgs:  nil,
+		RecoverArgs: nil,
+		RunTime:     time.Hour * time.Duration(7*24),
```

Comment:
`RunTime` means duration between injection and recovery this time, not means what's your understanding.